src/poetry/core/packages/dependency.py

```diff
@@ -167,7 +167,10 @@ def marker(self, marker: str | BaseMarker) -> None:
                         for _extra in extra_values:
                             if not _extra.startswith("!="):
                                 new_in_extras.append(canonicalize_name(_extra))
-            self._in_extras = [*self._in_extras, *new_in_extras]
+            self._in_extras = [
```

Comment:
Couldn't we use sets here to remove duplicates?

Reply:
That would change the order, and the order is important for determistic markers.

Reply:
Ok, suspected as much, but wanted to confirm (maybe we should consider vendoring something like orderedset?).